pac4j-saml/src/main/java/org/pac4j/saml/metadata/SAML2IdentityProviderMetadataResolver.java

```diff
@@ -71,27 +75,31 @@ public final MetadataResolver resolve() {
                     path = "/" + path;
                 }
                 resource = ResourceHelper.of(new ClassPathResource(path));
-            } else if (this.idpMetadataPath.startsWith("file:")) {
-                resource = ResourceHelper.of(new FileSystemResource(this.idpMetadataPath));
-            } else if (this.idpMetadataPath.startsWith("http")) {
+            }  else if (this.idpMetadataPath.startsWith("https")) {
```

Comment:
I would go for the easiest solution: allow HTTP and maybe add a warning if the url is in HTTP, but your proposal makes sense as well even if it's more work.